test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,867 @@
+			Expect(err).ToNot(HaveOccurred(), "failed to fetch mcp")
+			// Save the original performance profile
+			initialProfile = profile.DeepCopy()
+
+			for _, node := range workerRTNodes {
+				numaInfo, err := nodes.GetNumaNodes(&node)
+				Expect(err).ToNot(HaveOccurred())
+				if len(numaInfo) < 2 {
+					Skip(fmt.Sprintf("This test need 2 Numa nodes. The number of numa nodes on node %s < 2", node.Name))
+				}
+			}
+
+			By("Modifying Profile")
+			for _, node := range workerRTNodes {
+				numaCoreSiblings, err = nodes.GetCoreSiblings(&node)
+			}
+
+			// Get cpu siblings from core 0, 1
+			for reservedCores := 0; reservedCores < 2; reservedCores++ {
+				cpusiblings := nodes.GetAndRemoveCpuSiblingsFromMap(numaCoreSiblings, reservedCores)
+				reserved = append(reserved, cpusiblings...)
+			}
+			reservedCpus := strings.Join(reserved, ",")
+
+			for key := range numaCoreSiblings {
+				if numaZone[key] == nil {
+					numaZone[key] = make(map[int][]string)
+				}
+				for k := range numaCoreSiblings[key] {
+					cpusiblings := nodes.GetAndRemoveCpuSiblingsFromMap(numaCoreSiblings, k)
+					isolated = append(isolated, cpusiblings...)
+					numaZone[key][k] = append(numaZone[key][k], cpusiblings...)
+				}
+			}
+
+			// save the assigned cpus to isolated in a map based on zone and core
+			// we need to know which and how many cpus are available after cpus
+			// are assigned to reserved.
+			//Get available cpus in numa node 0 and numa node 1
+			for core := range numaZone[0] {
+				available_node0_cpus = append(available_node0_cpus, numaZone[0][core]...)
+			}
+
+			for core := range numaZone[1] {
+				available_node1_cpus = append(available_node1_cpus, numaZone[1][core]...)
+			}
+
+			isolatedCpus := strings.Join(isolated, ",")
+			reservedSet := performancev2.CPUSet(reservedCpus)
+			isolatedSet := performancev2.CPUSet(isolatedCpus)
+
+			//Enable Hugepages
+			hpSize2M := performancev2.HugePageSize("2M")
+			hpSize1G := performancev2.HugePageSize("1G")
+			profile.Spec.HugePages = &performancev2.HugePages{
+				DefaultHugePagesSize: &hpSize1G,
+				Pages: []performancev2.HugePage{
+					{
+						Count: int32(numaZone0HugepagesCount),
+						Size:  hpSize2M,
+						Node:  pointer.Int32Ptr(0),
+					},
+					{
+						Count: int32(numaZone1HugepagesCount),
+						Size:  hpSize1G,
+						Node:  pointer.Int32Ptr(1),
+					},
+				},
+			}
+			if !*profile.Spec.RealTimeKernel.Enabled {
+				profile.Spec.RealTimeKernel = &performancev2.RealTimeKernel{
+					Enabled: pointer.BoolPtr(true),
+				}
+			}
+			if profile.Spec.WorkloadHints == nil {
+				profile.Spec.WorkloadHints = &performancev2.WorkloadHints{
+					RealTime: pointer.Bool(true),
+				}
+			}
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reservedSet,
+				Isolated: &isolatedSet,
+			}
+			profile.Spec.NUMA = &performancev2.NUMA{
+				TopologyPolicy: &policy,
+			}
+			By("Updating Performance profile")
+			profiles.UpdateWithRetry(profile)
```

Comment:
can update the profile here using patch? I still haven't checked that, but from offline conversations, I understood that even using UpdateWithRetry may result in an error of that the object is already being modified..

Reply:
No we can't use the patch operation as i still could not reproduce the issue between patch and updateWithRetry

Reply:
ack